File: Frontend/LeaveManagement/src/app/features/LeaveRequest/dtos/LeaveRequestStatusResponseDto.ts
export interface LeaveRequestStatusResponseDto {
  leaveId: string;     // GUIDs are always strings in TypeScript/JavaScript for API responses
  employeeId: number;
  firstName?: string;
  lastName?: string;
  leaveType: string;
  startDate: Date;     // Changed to Date object
  endDate: Date;       // Changed to Date object
  totalDays: number;
  status: string;
  timeStamp: Date;     // Changed to Date object
}

File: Backend/LeaveManagement/LeaveManagement/Application/DTOs/LeaveRequestStatusResponseDTO.cs
namespace LeaveManagement.Application.DTOs;
public class LeaveRequestStatusResponseDTO
{
    public Guid LeaveId { get; set; }
    public int EmployeeId { get; set; }
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public required string LeaveType { get; set; }
    public DateOnly StartDate { get; set; }
    public DateOnly EndDate { get; set; }
    public int TotalDays { get; set; }
    public required string Status { get; set; }
    public DateTime TimeStamp { get; set; }
}
